src/test/java/org/broadinstitute/consent/http/db/StudyDAOTest.java

```diff
@@ -0,0 +1,279 @@
+                UUID.randomUUID()
+        );
+
+        Integer prop1Id = studyDAO.insertStudyProperty(
+                id,
+                "prop1",
+                PropertyType.String.toString(),
+                "asdf"
+        );
+
+        Integer prop2Id = studyDAO.insertStudyProperty(
+                id,
+                "prop2",
+                PropertyType.Number.toString(),
+                "1"
+        );
+
+        // create some random, other property
+        studyDAO.insertStudyProperty(
+                id2,
+                "unrelated",
+                PropertyType.String.toString(),
+                "asdfasdfasdf"
+        );
+
+        Study study = studyDAO.findStudyById(id);
+
+        assertEquals(study.getProperties().size(), 2);
+
+        study.getProperties().forEach((prop) -> {
+            if (prop.getStudyPropertyId().equals(prop1Id)) {
+                assertEquals("prop1", prop.getName());
+                assertEquals(PropertyType.String, prop.getType());
+                assertEquals("asdf", prop.getValue());
+            } else if (prop.getStudyPropertyId().equals(prop2Id)) {
+                assertEquals("prop2", prop.getName());
+                assertEquals(PropertyType.Number, prop.getType());
+                assertEquals(1, prop.getValue());
+            } else {
+                fail("Unexpected property");
+            }
+        });
+    }
+
+
+    @Test
+    public void testAlternativeDataSharingPlan() {
+        User u = createUser();
+
+        UUID uuid = UUID.randomUUID();
+        Integer id = studyDAO.insertStudy(
+                "name",
+                "description",
+                "asdf",
+                List.of(),
+                true,
+                u.getUserId(),
+                Instant.now(),
+                uuid
+        );
+
+        FileStorageObject fso = createFileStorageObject(uuid.toString(), FileCategory.ALTERNATIVE_DATA_SHARING_PLAN);
+
+        Study study = studyDAO.findStudyById(id);
+
+        assertEquals(fso.getFileStorageObjectId(), study.getAlternativeDataSharingPlan().getFileStorageObjectId());
+        assertEquals(fso.getBlobId(), study.getAlternativeDataSharingPlan().getBlobId());
+
+    }
+
+    @Test
+    public void testIncludesDatasetIds() {
+        Study s = this.insertStudyWithProperties();
+
+        insertDataset();
+        Dataset ds1 = this.insertDatasetForStudy(s.getStudyId());
+        insertDataset();
+        Dataset ds2 = this.insertDatasetForStudy(s.getStudyId());
+        insertDataset();
+
+        s = studyDAO.findStudyById(s.getStudyId());
+
+        assertEquals(2, s.getDatasetIds().size());
+        assertTrue(s.getDatasetIds().contains(ds1.getDataSetId()));
+        assertTrue(s.getDatasetIds().contains(ds2.getDataSetId()));
+    }
+
+    private FileStorageObject createFileStorageObject(String entityId, FileCategory category) {
+        String fileName = org.apache.commons.lang3.RandomStringUtils.randomAlphabetic(10);
```

Comment:
```suggestion
        String fileName = randomAlphabetic(10);
```

I think these can be pulled out since you import the `RandomStringUtils`.